code/core/src/core-server/presets/common-preset.ts

```diff
@@ -288,3 +290,42 @@ export const managerEntries = async (existing: any) => {
     ...(existing || []),
   ];
 };
+
+// Store the promise (not the result) to prevent race conditions.
+// The promise is assigned synchronously, so concurrent calls will share the same initialization.
+// This is essentially an async singleton pattern.
+let storyIndexGeneratorPromise: Promise<StoryIndexGenerator> | undefined;
```

Comment:
does the `storyIndexGenerator` needs to be invalidated at some point?

Reply:
no, it is always the same instance.